### ISSUE TITLE
mdx tag is not resolved by components

### ISSUE THREAD

**Issue by SivertGullbergHansen** [contentlayerdev/contentlayer#389]
Hi 👋🏼 

## `<p>` tag is not resolved by custom component

If I want to style my mdx content with a `<p>` tag it is not considered by the `useMDXComponent`'s `component` property.

### Visual example

![image](https://user-images.githubusercontent.com/6785315/222933372-502459f7-bedb-4121-9849-9cf669ce1638.png)

### My code
```jsx
// MdxConvertedComponents.tsx
import { ArticleContentVariant } from "config/animations";
import { motion } from "framer-motion";

const Components: any = {};

function Custom({tag, children, ...props}: {tag: string, children: any, props: any}) {
    const C = motion[tag]
    return <C variants={ArticleContentVariant} {...props}>{ children }</C>
}

// ...

Components.p = (props) => {
  return <Custom className='text-primary' tag='p' {...props}>{props?.children}</Custom>;
};

export { Components };
```

```jsx
// [slug].tsx
import { Components } from "components/MdxConvertedComponents";
<MDXContent components={Components} />
```

It's strange that it works on `non-p` tags, but not on the actual `p` tag in the mdx file 🤔. I'd like to keep using the standard HTML syntax and tailwind styling in my mdx files rather than using custom components (like replacing `<p>` with `<P>`).

Any ideas?

**Comment by nkilm** [comment]
@SivertGullbergHansen Are you using `prose` classes from [@tailwindcss/typography](https://tailwindcss.com/docs/typography-plugin)? if yes then check if your custom styles are being overridden by typography's classes or not.

**Comment by SivertGullbergHansen** [comment]
> @SivertGullbergHansen Are you using `prose` classes from [@tailwindcss/typography](https://tailwindcss.com/docs/typography-plugin)? if yes then check if your custom styles are being overridden by typography's classes or not.

You are correct, I am styling my articles with Tailwind's prose. After removing all styling from the article page, the behavior has not changed. 🤔

It just doesn't seem like its parsing the `p` tag in my mdx file, like the `MDXContent` doesn't know what to do about the `p` tag so it leaves it as is 🤔 

If I define the `'p'` component as `'a'` in the components parameter, then everything BUT the `p` tag in my mdx file is changed to an `a` tag.
```jsx
// [slug].tsx
<MDXContent components={{...Components, 'p': 'a'}} />
```

```html
// first-commit.mdx
When you change a source file, Contentlayer automatically...

<p>Here's some text</p>

Nisi commodo in consectetur sit eu anim laborum magna velit magna...
```

Results in:
![image](https://user-images.githubusercontent.com/6785315/222986983-bdebf587-a43a-4655-92bd-777cd7efdd63.png)

@nkilm is the p-tag defined as something else in the components property?

**Comment by NekoChanTaiwan** [comment]
I think I also had a similar problem.

![image](https://user-images.githubusercontent.com/47508893/223166192-f6e6a6ea-2520-404f-908b-926bc4f7f466.png)

**Comment by NekoChanTaiwan** [comment]
> Yes this is similar to the issue I am having. Defining a tag with small letters, as both you and I have done, seems to not work 🤔

But in my case, my other HTML tags are working with components, only the kbd tag.
